packages/mattermost-redux/src/types/emojis.ts

```diff
@@ -1,38 +1,43 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-export type EmojiCategory = (
+export type EmojiCategory =
     | 'recent'
-    | 'people'
-    | 'nature'
-    | 'foods'
-    | 'activity'
-    | 'places'
+    | 'searchResults'
+    | 'smileys-emotion'
+    | 'people-body'
+    | 'animals-nature'
+    | 'food-drink'
+    | 'activities'
+    | 'travel-places'
     | 'objects'
     | 'symbols'
     | 'flags'
-    | 'custom'
-);
+    | 'custom';
+
 export type CustomEmoji = {
     id: string;
+    name: string;
+    category: 'custom';
     create_at: number;
     update_at: number;
     delete_at: number;
     creator_id: string;
-    name: string;
-    category: 'custom';
 };
 export type SystemEmoji = {
     name: string;
+    category: EmojiCategory;
     image: string;
     short_name: string;
     short_names: string[];
-    category: EmojiCategory;
     batch: number;
     skins?: string[];
-    skin_variations?: unknown[]; // we currently don't have a use for it other than knowing the field exists.
+    skin_variations?: Record<string, Emoji>; // we currently don't have a use for it other than knowing the field exists.
+    unified?: string;
 };
-export type Emoji = SystemEmoji | CustomEmoji;
+
+export type Emoji = Partial<Omit<CustomEmoji, 'name' | 'category'>> & Partial<Omit<SystemEmoji, 'name' | 'category'>> & Pick<SystemEmoji, 'name' | 'category'> & { filename?: string };
```

Comment:
Similarly to `CategoryOrEmojiRow`, I think we should be using a union type here with type predicates (like the ones in https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/utils/emoji_utils.ts` but updated to use the new type definitions here)

Reply:
✅  Updated as per request/discussion